### DISCUSSION TITLE
Cloudflare workers, always response with Http Code 200, no matter the status I set in the response with c.json

### DISCUSSION THREAD

**Discussion by Makisuo** [honojs/hono#1174] [Q&A]
`return c.json({
			body: {
				status: "404",
				message: "Project doesn't exist",
			},
			status: 404,
		})`
		
		what am I doing wrong

**Comment by harshagarwalnyu** [comment]
The issue is that `c.json()` sets the status code separately from the body. Passing `status: 404` inside the body object does not set the HTTP status -- it just adds a field to the JSON payload.

Fix: pass the status code as the second argument to `c.json()`:

```ts
return c.json(
  {
    body: {
      status: "404",
      message: "Project doesn't exist",
    },
  },
  404  // <-- HTTP status code goes here
)
```

The signature is `c.json(data, statusCode?)`. Without the second argument it defaults to 200.
